corehq/apps/user_importer/importer.py

```diff
@@ -578,3 +580,127 @@ def _get_domain_info(domain):
             ret['errors'].append(_error_message)
 
     return ret
+
+
+def create_or_update_web_users(upload_domain, user_specs, upload_user, update_progress=None):
+
+    ret = {"errors": [], "rows": []}
+    current = 0
+
+    try:
+        for row in user_specs:
+            if update_progress:
+                update_progress(current)
+                current += 1
+            role_updated = False
+
+            username = row.get('username')
```

Comment:
do we ensure that these are valid emails (a requirement for web user usernames)

Reply:
Yes one of the validators is an email validator